compiler/modules/front.syntax/scan.bal

```diff
@@ -567,7 +568,7 @@ function scanNormal(int[] codePoints, int startIndex, Scanned result) {
                     i = scanIdentifier(codePoints, i + 1, result);
                 }
                 else {
-                    endFragment(CP_UNDERSCORE, i, result);
+                    endIdentifierOrKeyword(codePoints, i, result);
```

Comment:
Just call endFragment here.  (You know what the keyword index is.)

Reply:
Fixed